keps/0019-package-api-versioning.md

```diff
@@ -202,7 +207,8 @@ Two proposals: singular and composite package/operator version. In the case of p
 ##### Cons
 
 - Visually more complex
-- Makes it complex to know which operator-related features are available in which versions
+- Makes it complex to know which operator-related features are available in which versions.  Operator revision for different app versions are unrelated
+- Upgrades paths are unclear. Upgrading from `3.11.4-0.1.2` to `3.12.0-0.1.2` might result in missing some features that were present in `3.11.4-0.1.2`
```

Comment:
> Upgrades paths are unclear

The same is true for single version afaict. The rest of the sentence is fair.

In practice, in the example given above it would be more likely that `3.12.0-0.1.2` would have more features then `3.11.4-0.1.2`, even though it will still be theoretically possible that the later app version doesn't have a feature from an earlier app version.

Reply:
this is not the same for single semver. Where the development is continuous. Here with operator revision reset. It's hard to predict if `3.11.4-0.1.2` to `3.12.0-0.1.2` is just base tech bump or features bump. Or if some features are introduced or same feature parity will be in `3.12.0-0.1.3` or `3.12.0-0.1.1`

Reply:
Yep, that's what I meant with "The rest of the sentence is fair." The "Upgrades paths are unclear" part is also for the single version afaict, just in a different way.

Reply:
Is this suggestion fair? https://github.com/kudobuilder/kudo/pull/1103#discussion_r350165019

Reply:
yeah, I will add that,  I didn't have given enough time to go through all cons, as most of the cons of a single version also apply for the composite version.
`Operator users care about the application version, not a "synthetic" version` can be translated to `Operator users care about the operator features also` 
And `it might be possible that subsequent package versions with different app versions don't contain an operator-related feature or bug fix that only makes sense for a specific app version` is true for composite version also. `3.11.4-0.1.2` might have a feature that `3.12.0-0.1.2` don't have as that was specific to `3.11.4`